app/factories/bulkrax/object_factory_decorator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Bulkrax
+  module ObjectFactoryDecorator
+    def find(id)
+      super
+    rescue Ldp::BadRequest => e
+      Rails.logger.warn "Bulkrax::ObjectFactory: Error finding object with id #{id}: #{e.message}"
+      Rails.logger.warn "Trying to find by bulkrax_identifier_tesim instead."
+      ActiveFedora::Base.where(bulkrax_identifier_tesim: id).first
```

Comment:
The [base Bulkrax object factory interface](https://github.com/samvera/bulkrax/blob/main/app/factories/bulkrax/object_factory_interface.rb#L323-L325) does this by default, but [is overwritten not to](https://github.com/samvera/bulkrax/blob/main/app/factories/bulkrax/object_factory.rb#L60-L64) for `Bulkrax::ObjectFactory` 🤔 The intention seems to be to explicitly not fall back on `source_identifier`, but I don't know why. I think it's worth double checking this override with @ShanaLMoore or someone else involved with that change